mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSchemaReader.java

```diff
@@ -48,12 +48,21 @@ public Schema getSchema(ResultSetMetaData metadata, int index) throws SQLExcepti
     int sqlType = metadata.getColumnType(index);
     String sqlTypeName = metadata.getColumnTypeName(index);
 
-    // YEAR type in MySQL should get converted to integer to avoid truncation
-    // failures in the MySQL Sink plugin due to missing date and month details.
-    if (sqlType == Types.DATE && YEAR_TYPE_NAME.equalsIgnoreCase(sqlTypeName)) {
-      return Schema.of(Schema.Type.INT);
+    switch(sqlType) {
+      case Types.DATE:
+        // YEAR type in MySQL should get converted to integer to avoid truncation
+        // failures in the MySQL Sink plugin due to missing date and month details.
+        if (YEAR_TYPE_NAME.equalsIgnoreCase(sqlTypeName)) {
+          return Schema.of(Schema.Type.INT);
+        }
+        break;
+      case Types.INTEGER:
+        // MEDIUMINT UNSIGNED should be mapped to int
```

Comment:
why is MEDIUMINT UNSIGNED a special case, shouldn't a signed mediumint also be an int?

Reply:
In the DBUtils we have a central logic of converting all Int unsigned types to long. But since Mysql Mediumint unsigned type have same sql type as that of int, hence it also gets converted to long type. This cause a failure in BigQuery sink during the serialization to json type. Since int can contain the mediumint unsigned min and max value hence it should ideally be mapped to int.

Reply:
ok I see. It would be good to put that in the code comment.